spec/models/video_spec.rb

```diff
@@ -37,5 +38,120 @@
+    end
+  end
+  describe "sum_review_ratings" do
+    it "should return 0 if there are no reviews" do
+      video = Fabricate(:video)
+      expect(video.sum_review_ratings).to eq(0)
+    end
+    it "should return 4 if there is one review with a rating of 4" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video, rating: 4)
+      expect(video.sum_review_ratings).to eq(4)
+    end
+    it "should return 5 if there are two reviews with a rating of 2 and 3" do
+      video = Fabricate(:video)
+      [2, 3].each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.sum_review_ratings).to eq(5)
+    end
+    it "should return 15 if there are 5 reviews with ratings of 1,2,3,4, and 5" do
+      video = Fabricate(:video)
+      (1..5).each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.sum_review_ratings).to eq(15)
+    end
+  end
+  describe "average_rating_of_reviews" do 
+    it "returns nil if there are no reviews" do
+      video = Fabricate(:video)
+      expect(video.average_rating_of_reviews).to be_nil
+    end
+    it "returns 1.0 if there is only one review with a rating of 1" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video, rating: 1)
+      expect(video.average_rating_of_reviews).to eq(1.0)
+    end
+    it "returns 5.0 if there is only one review with a rating of 5" do
+      video = Fabricate(:video)
+      Fabricate(:review, video: video, rating: 5)
+      expect(video.average_rating_of_reviews).to eq(5.0)
+    end
+    it "returns 5.0 if there are 10 reviews with a rating of 5 each" do
+      video = Fabricate(:video)
+      10.times {Fabricate(:review, video: video, rating: 5)}
+      expect(video.average_rating_of_reviews).to eq(5.0)
+    end
+    it "returns 1.0 if there are 20 reviews with a rating of 1 each" do
+      video = Fabricate(:video)
+      10.times {Fabricate(:review, video: video, rating: 1)}
+      expect(video.average_rating_of_reviews).to eq(1.0)
+    end
+    it "returns 3.5 if there are 2 reviews with a rating of 3 and 4 respectively" do
+      video = Fabricate(:video)
+      [3,4].each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.average_rating_of_reviews).to eq(3.5)
+    end
+    it "returns 3.25 if there are 4 reviews with a rating of 3, 3, 3, and 4 respectively" do
+      video = Fabricate(:video)
+      [3,3,3,4].each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.average_rating_of_reviews).to eq(3.3)
+    end
+    it "returns 3.33 if there are 3 reviews with a rating of 3, 3, and 4 respectively" do
+      video = Fabricate(:video)
+      [3,3,4].each do |i|
+        Fabricate(:review, video: video, rating: i)
+      end
+      expect(video.average_rating_of_reviews).to eq(3.3)
+    end
+    it "returns 3.67 if there are 3 reviews with a rating of 3, 4, and 4 respectively" do
```

Comment:
I think you are over-testing these scenarios. Same comment as above, each test created overhead and takes time. I think you could use half of these fraction tests and be fine.